src/Sylius/Bundle/ApiBundle/Filters/TranslationOrderNameFilter.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Filters;
+
+use ApiPlatform\Core\Bridge\Doctrine\Common\Filter\OrderFilterInterface;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Filter\AbstractContextAwareFilter;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use Doctrine\ORM\QueryBuilder;
+
+final class TranslationOrderNameFilter extends AbstractContextAwareFilter
+{
+    protected function filterProperty(
+        string $property,
+        $value,
+        QueryBuilder $queryBuilder,
+        QueryNameGeneratorInterface $queryNameGenerator,
+        string $resourceClass,
+        string $operationName = null
+    ): void {
+        if ('order' === $property) {
+            if (!isset($value['translation.name'])) {
+                return;
+            }
+
+            $direction = $value['translation.name'];
+
+            if (isset($value['localeCode'])) {
+                $queryBuilder
+                    ->addSelect('translation')
+                    ->innerJoin(
+                        'o.translations',
```

Comment:
We cannot be sure, that `o` will be used here.
```suggestion
                        sprintf('%s.translations', $queryBuilder->getRootAliases()[0]));
```